selfdrive/controls/lib/vehicle_model.py

```diff
@@ -163,10 +163,11 @@ def kin_ss_sol(sa: float, u: float, VM: VehicleModel) -> np.ndarray:
   Returns:
     2x1 matrix with steady state solution
   """
-  K = np.zeros((2, 1))
+  K: np.ndarray = np.zeros((2, 1))
   K[0, 0] = VM.aR / VM.sR / VM.l * u
   K[1, 0] = 1. / VM.sR / VM.l * u
-  return K * sa
+  K_scaled: np.ndarray = K * sa  
```

Comment:
Same here, it can't infer?

Reply:
It really can't :( Not sure why, it knows K is `np.ndarray` and `sa` is float but can't prove that the product is also an `np.ndarray`.

Reply:
@GregorKikelj re: numpy support https://github.com/commaai/openpilot/pull/24379#issuecomment-1113969757

I was able to get numpy type checking enabled by adding `plugins = numpy.typing.mypy_plugin` to [mypy.ini](https://github.com/commaai/openpilot/blob/master/mypy.ini) and adding `'numpy'` to additional_dependencies:

https://github.com/commaai/openpilot/blob/ec7cd74dc37caad9a281f70fa1bfcb12535efeb9/.pre-commit-config.yaml#L23

Note that typing for shapes is still a WIP (https://github.com/numpy/numpy/issues/16544).

Reply:
Will check on monday, need to get to correct PC :) Thanks for the hint.